cmd/kubeadm/app/cmd/phases/reset/unmount_linux.go

```diff
@@ -24,30 +24,52 @@ import (
 	"strings"
 	"syscall"
 
-	"k8s.io/klog/v2"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 )
 
+var flagMap = map[string]int{
+	kubeadmapi.UnmountFlagMNTForce:       unix.MNT_FORCE,
+	kubeadmapi.UnmountFlagMNTDetach:      unix.MNT_DETACH,
+	kubeadmapi.UnmountFlagMNTExpire:      unix.MNT_EXPIRE,
+	kubeadmapi.UnmountFlagUmountNoFollow: unix.UMOUNT_NOFOLLOW,
+}
+
+func flagsToInt(flags []string) int {
+	res := 0
+	for _, f := range flags {
+		res |= flagMap[f]
+	}
+	return res
+}
+
 // unmountKubeletDirectory unmounts all paths that contain KubeletRunDirectory
-func unmountKubeletDirectory(absoluteKubeletRunDirectory string) error {
+func unmountKubeletDirectory(kubeletRunDirectory string, flags []string) error {
 	raw, err := os.ReadFile("/proc/mounts")
 	if err != nil {
 		return err
 	}
 
-	if !strings.HasSuffix(absoluteKubeletRunDirectory, "/") {
+	if !strings.HasSuffix(kubeletRunDirectory, "/") {
 		// trailing "/" is needed to ensure that possibly mounted /var/lib/kubelet is skipped
-		absoluteKubeletRunDirectory += "/"
+		kubeletRunDirectory += "/"
 	}
 
+	var errList []error
 	mounts := strings.Split(string(raw), "\n")
 	for _, mount := range mounts {
 		m := strings.Split(mount, " ")
-		if len(m) < 2 || !strings.HasPrefix(m[1], absoluteKubeletRunDirectory) {
+		if len(m) < 2 || !strings.HasPrefix(m[1], kubeletRunDirectory) {
 			continue
 		}
-		if err := syscall.Unmount(m[1], 0); err != nil {
-			klog.Warningf("[reset] Failed to unmount mounted directory in %s: %s", absoluteKubeletRunDirectory, m[1])
+		if err := syscall.Unmount(m[1], flagsToInt(flags)); err != nil {
+			errList = append(errList, errors.Wrapf(err, "failed to unmount %s", m[1]))
```

Comment:
```suggestion
			errList = append(errList, fmt.Errorf("failed to unmount %s: %w", m[1], err))
```
We will have stack trace below, maybe we don't need this `Wrapf` here

Reply:
we don't use %w and fmt.Errorf in kubeadm, intentionally

Reply:
OK, that's a little bit strange. Then how about errors.WithMessagef ?

Reply:
if we only care about annotating the error, like in this case using WithMessage is more correct, i can update this.
FWIW, we do use consecutive Wraps all over kubeadm within a call stack, even if it is more accurate to put Wraps only in key locations.

Reply:
updated